RePak/src/assets/material.cpp

```diff
@@ -167,77 +183,114 @@ void Assets::AddMaterialAsset(std::vector<RPakAssetEntryV7>* assetEntries, const
+    if (type == "gen")
     {
-        // I HAVE PROBABLY BROKEN THIS AT SOME POINT - spoon
 
-        //for testing purposes ""sknp"" will be a material with a full suite of textures.
+        if (subtype == "loadscreen") {
 
-        // These should always be constant (per each material type)
-        // There's different versions of these for each material type
-        // GUIDRefs[4] is Colpass entry.
+            mtlHdr->Flags2 = 0x10000002;
 
-        //apex default shader
-        mtlHdr->GUIDRefs[0] = 0x2B93C99C67CC8B51;
-        mtlHdr->GUIDRefs[1] = 0x1EBD063EA03180C7;
-        mtlHdr->GUIDRefs[2] = 0xF95A7FA9E8DE1A0E;
-        mtlHdr->GUIDRefs[3] = 0x227C27B608B3646B;
+            mtlHdr->ShaderSetGUID = 0xA5B8D4E9A3364655;
 
-        mtlHdr->GUIDRefs[0] = 0x39C739E9928E555C;
-        mtlHdr->GUIDRefs[1] = 0x67D89B36EDCDDF6E;
-        mtlHdr->GUIDRefs[2] = 0x43A9D8D429698B9F;
+        }
+        else {
+
+            Warning("Invaild type used! Defaulting to subtype 'loadscreen'... \n");
+
+            mtlHdr->Flags2 = 0x10000002;
+
+            mtlHdr->ShaderSetGUID = 0xA5B8D4E9A3364655;
+
+        }
+
+    // These should always be constant (per each material type)
+    // There's different versions of these for each material type\
+        // GUIDRefs[3] is Colpass entry, however loadscreens do not have colpass materials.
+
+        mtlHdr->GUIDRefs[0] = 0x0000000000000000;
+        mtlHdr->GUIDRefs[1] = 0x0000000000000000;
+        mtlHdr->GUIDRefs[2] = 0x0000000000000000;
+
+        mtlHdr->Flags = 0x00050300;
+
+        mtlHdr->unk6 = 0xFBA63181;
+
+    }
+    else if (type == "wld")
+    {
+        Warning("Type 'wld' is not supported currently!!!");
+        return;
+        /*
+        // THIS IS 'wld' IN TITANFALL (I think)
+
+        //UNSUPPORTED CURRENTLY
+
+        // GUIDRefs[4] is Colpass entry which is optional for wldc.
+        mtlHdr->GUIDRefs[0] = 0x435FA77E363BEA48; // DepthShadow
+        mtlHdr->GUIDRefs[1] = 0xF734F96BE92E0E71; // DepthPrepass
+        mtlHdr->GUIDRefs[2] = 0xD306370918620EC0; // DepthVSM
+        mtlHdr->GUIDRefs[3] = 0xDAB17AEAD2D3387A; // DepthShadowTight
 
         RePak::RegisterGuidDescriptor(subhdrinfo.index, offsetof(MaterialHeaderV12, GUIDRefs));
         RePak::RegisterGuidDescriptor(subhdrinfo.index, offsetof(MaterialHeaderV12, GUIDRefs) + 8);
         RePak::RegisterGuidDescriptor(subhdrinfo.index, offsetof(MaterialHeaderV12, GUIDRefs) + 16);
+        RePak::RegisterGuidDescriptor(subhdrinfo.index, offsetof(MaterialHeaderV12, GUIDRefs) + 24);
 
-        RePak::AddFileRelation(assetEntries->size(), 3);
-        assetUsesCount += 3;
+        //RePak::AddFileRelation(assetEntries->size(), 4);
+        //assetUsesCount += 4;
 
-        //mtlHdr->ShaderSetGUID = 0x1D9FFF314E152725;
-        mtlHdr->ShaderSetGUID = 0x586783F71E99553D;
+        mtlHdr->ShaderSetGUID = 0x4B0F3B4CBD009096;
+        */
     }
-    */
-
-    if (type == "skn")
+    else if (type == "fix")
```

Comment:
Any particular reason why we have changed from using skn as the type?

Reply:
Better sorting, 'skn' is still used. In vanilla rpaks/Titanfall 2 most materials have a 'fix' version, they use a different set of shaders and a different shaderset, I'm not sure what they do exactly but I've added it just in case they are needed in the future.